esm_parser/yaml_to_dict.py

```diff
@@ -66,16 +79,61 @@ def constructor_env_variables(loader, node):
         :return: the parsed string that contains the value of the environment
         variable
         """
+        # file name (including the full path if not in PWD)
+        fname = node.start_mark.name
+
+        # line number (starts from 0) where match is found
+        line_num = node.start_mark.line
+
+        # start and end of the colums numbers in the line (starts from 0)
+        # For character index from the start of the file use node.start_mark.index
+        column_start = node.start_mark.column
+        column_end = node.end_mark.column
+        
+        # read the file into a list of line. This can also be achieved by 
+        # read() instead but since the YAML files are not large (max few KBs),
+        # this is a quicked solution
+        file_lines = open(fname, 'r').readlines()
+        
+        # current line without the newline at the end 
+        cur_line = file_lines[line_num].rstrip()
```

Comment:
Here the file is opened but is not closed later. As far as I understand this is recommended against. I typically prefer to use a `with open(<path>, 'r') as <file>:` cause then I don't have to remember to close the file later.

Reply:
Basically, the Python garbage collector would take care about that but I agree that this is bit lazy writing. I will separate that into open(), readlines() and close().